azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/WarArtifactHandlerImpl.java

```diff
@@ -39,15 +39,19 @@ public void publish(IDeployTargetAdapter deployTarget) throws MojoExecutionExcep
         final String path = getContextPath();
 
         int retryCount = 0;
-        mojo.getLog().info("Starting to deploy the war file...");
+        mojo.getLog().info("Deploying the war file...");
         while (retryCount++ < DEFAULT_MAX_RETRY_TIMES) {
             try {
                 deployTarget.warDeploy(war, path);
                 return;
             } catch (Exception e) {
-                mojo.getLog().warn(String.format(UPLOAD_FAILURE, retryCount, DEFAULT_MAX_RETRY_TIMES));
+                mojo.getLog().warn(String.format(UPLOAD_FAILURE, e.getMessage(), retryCount, DEFAULT_MAX_RETRY_TIMES));
             }
         }
+
+        if (retryCount > DEFAULT_MAX_RETRY_TIMES) {
```

Comment:
Can we add this cases into UT?

Reply:
UT that tests if the expected exception get thrown? sounds like a good idea

Reply:
Yes